### ISSUE TITLE
Support for proxy reset

### ISSUE THREAD

**Issue by strekm** [kyma-project/istio#66]
**Description**

Enhance Istio operator with support for proxy reset. Proxies should be up to date after Istio upgrade. Supported workloads should be the same as implementation in Istio reconciler. There's no need to analyse cluster looking for workloads in service mesh but not having sidecars (this was fixed by istio). Workloads should not be annotated nor labeled. Configuration of sidecar resources is out of scope of this task.

- [x] sidecars are up to date after Istio upgrade
- [x] workloads ONLY have annotation `restarted at`
- [x] workloads have no labels added

**Reasons**

modularisation

**DoD:**
~~- [ ] provide documentation~~
- [x] provide unit tests
~~- [ ] test on SKR~~
~~- [ ] verify resource limits~~
- [x] followup issue

**Attachments**
depends on: https://github.com/kyma-project/istio/issues/55
part of: https://github.com/kyma-project/istio/issues/38
